apps/condo/domains/news/schema/ExportNewsRecipientsService.js

```diff
@@ -0,0 +1,242 @@
+        replaces: {
+            header: HeaderMessage,
+            newsRecipients: processedRows,
+            i18n: {
+                ...getHeadersTranslations(EXPORT_TYPE_NEWS_RECIPIENTS, locale),
+                sheetName: i18n('excelExport.sheetNames.newsRecipients', { locale }),
+            },
+        },
+    })
+
+    return linkToFile
+}
+
+/**
+ * List of possible errors, that this custom schema can throw
+ * They will be rendered in documentation section in GraphiQL for this custom schema
+ */
+const errors = {
+    MISSING_PROPERTY: {
+        mutation: 'exportNewsRecipients',
+        code: INTERNAL_ERROR,
+        type: NOT_FOUND,
+        message: 'Could not found property related to newsItemScope',
+        messageForUser: 'api.user.exportNewsRecipients.MISSING_PROPERTY',
+    },
+    WRONG_ORGANIZATION_ID: {
+        mutation: 'exportNewsRecipients',
+        code: INTERNAL_ERROR,
+        type: NOT_FOUND,
+        message: 'Could not found organization by provided id',
+        messageForUser: 'api.user.exportNewsRecipients.WRONG_ORGANIZATION_ID',
+    },
+}
+
+const ExportNewsRecipientsService = new GQLCustomSchema('ExportNewsRecipientsService', {
+    types: [
+        {
+            access: true,
+            type: 'input ExportNewsRecipientsInput { dv: Int!, sender: JSON!, organizationId: ID!, newsItemScopes: [NewsItemScopeWhereInput] }',
+        },
+        {
+            access: true,
+            type: 'type ExportNewsRecipientsOutput { status: String!, linkToFile: String! }',
+        },
+    ],
+    
+    mutations: [
+        {
+            access: access.canExportNewsRecipients,
+            schema: 'exportNewsRecipients(data: ExportNewsRecipientsInput!): ExportNewsRecipientsOutput',
+            resolver: async (parent, args, context, info, extra = {}) => {
+                const { data: { newsItemScopes, organizationId } } = args
+                const locale = extractReqLocale(context.req) || conf.DEFAULT_LOCALE
+
+                const organization = await Organization.getOne(context, {
+                    id: organizationId,
+                    deletedAt: null,
+                })
+                if (!organization) throw new GQLError(errors.WRONG_ORGANIZATION_ID, context)
+
+                let properties
+                if (!isEmpty(newsItemScopes)) {
+                    properties = newsItemScopes.map(newsItemScope => {
+                        if (!isNil(newsItemScope.property)) return newsItemScope.property.id
+                    })
+                }
+
+
+                let residentsByProperties
+                if (!isEmpty(properties)) {
+                    residentsByProperties = await loadListByChunks({
+                        context: context,
+                        list: Resident,
+                        chunkSize: 50,
+                        limit: 100000,
+                        where: {
+                            property: {
+                                id_in: properties,
```

Comment:
Some elements of the `properties` array may be `undefined`. It would be good to filter such values.